python/_tskitmodule.c

```diff
@@ -10135,24 +10135,71 @@ TreeSequence_pair_coalescence_quantiles(
     return ret;
 }
 
+static PyArrayObject *
+parse_sites(TreeSequence *self, PyObject *sites, npy_intp *out_dim)
+{
+    PyArrayObject *array;
+    tsk_size_t num_sites = tsk_treeseq_get_num_sites(self->tree_sequence);
+
+    if (sites == Py_None) {
+        array = (PyArrayObject *) PyArray_Arange(0, num_sites, 1, NPY_INT32);
+        if (array == NULL) {
+            goto out;
+        }
+        *out_dim = PyArray_DIM(array, 0);
+    } else {
+        array = (PyArrayObject *) PyArray_FROMANY(
+            sites, NPY_INT32, 1, 1, NPY_ARRAY_IN_ARRAY);
+        if (array == NULL) {
+            goto out;
+        }
+        *out_dim = PyArray_DIM(array, 0);
+    }
+
+out:
+    return array;
+}
+
+static PyArrayObject *
+parse_positions(TreeSequence *self, PyObject *positions, npy_intp *out_dim)
+{
+    PyArrayObject *array;
+
+    if (positions == Py_None) {
+        array = (PyArrayObject *) TreeSequence_get_breakpoints(self);
+        if (array == NULL) {
+            goto out;
+        }
+        *out_dim = PyArray_DIM(array, 0) - 1; // NB the last element must be truncated
+    } else {
+        array = (PyArrayObject *) PyArray_FROMANY(
+            positions, NPY_FLOAT64, 1, 1, NPY_ARRAY_IN_ARRAY);
+        if (array == NULL) {
```

Comment:
Should cover this in test_lowlevel.py by passing in an bad array, say one with the wrong or the wrong type.

Reply:
I missed that case, I've added some tests that cover it